lib/_stream_readable.js

```diff
@@ -570,7 +570,8 @@ function maybeReadMore(stream, state) {
 function maybeReadMore_(stream, state) {
   var len = state.length;
```

Comment:
This is somewhat outside the scope of this change, but given I'm touching the function, I'll raise it here anyway.

The `len` variable. Is there a reason it is introduced outside the loop and thus needs to be updated at the end of the loop for the next iteration? Maybe some performance considerations by declaring the variable outside the loop?

```javascript
var len = state.length  // Value stored before the loop begins.
while( /* does not use 'len' */ ) {
  stream.read(0);  // May update 'state.length'
  if (len === state.length)
    break;
  else
    len = state.length;  // We'll need to update 'len' for the next iteration.
}
```

Moving the `len` fully inside the loop would remove the need to maintain its state between different loop iterations:

```javascript
while( /* does not use 'len' */ ) {
  const len = state.length  // Store 'len' inside the loop.
  stream.read(0);  // May update 'state.length'
  if (len === state.length)
    break;
  // No need to update 'len' here. If we loop again, we'll refresh 'len' before the call to read().
}
```

This is annoying me enough that I'm willing to make the change (as a separate PR if that's the wish). But I also understand if you'd rather keep the noise of "stylistic" changes to minimum and would rather keep the code as it is.

Reply:
Feel free to do this as part of that change.

Reply:
Made the change. I'm assuming 'as part of that change' was referring to this PR.